File: 203. Remove Linked List Elements/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    headNode := new(ListNode)   // important eg: [1, 2] 1
    headNode.Next = head
    cur := head
    pre := headNode
    for {
        if cur == nil {
            break
        }
        if cur.Val == val {
            pre.Next = cur.Next
            cur = cur.Next // [1] 1 is wrong
        } else {
            pre = cur
            cur = cur.Next
        }
    }
    return headNode.Next
}

File: 203. Remove Linked List Elements/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let headNode = new ListNode()
    headNode.next = head
    pre = headNode
    cur = head
    while(cur) {
        if(cur.val === val) {
            pre.next = cur.next
        } else {
            pre = cur
        }
        cur = cur.next
    }
    return headNode.next
};